attest/eventlog.go

```diff
@@ -741,3 +748,73 @@ func parseRawEvent2(r *bytes.Buffer, specID *specIDEvent) (event rawEvent, err e
 	}
 	return event, err
 }
+
+// CombineEventlogs takes a series of TPM 2.0 event logs and combines
+// them into a single sequence.
+//
+// Additional logs must not use a digest algorithm which was not
+// present in the original log.
+func CombineEventlogs(base []byte, additional ...[]byte) ([]byte, error) {
+	baseLog, err := ParseEventLog(base)
+	if err != nil {
+		return nil, fmt.Errorf("base: %v", err)
+	}
+	if baseLog.specIDEvent == nil {
+		return nil, errors.New("tpm 1.2 event logs cannot be combined")
+	}
+
+	outBuff := make([]byte, len(base))
+	copy(outBuff, base)
+	out := bytes.NewBuffer(outBuff)
+
+	for i, l := range additional {
+		log, err := ParseEventLog(l)
+		if err != nil {
+			return nil, fmt.Errorf("log %d: %v", i, err)
+		}
+		if log.specIDEvent == nil {
+			return nil, fmt.Errorf("log %d: cannot use tpm 1.2 event log as a source", i)
+		}
+
+	algCheck:
```

Comment:
nit: just use an if statement instead of a tagged continue

```
		for _, alg := range log.specIDEvent.algs {
			algOk := false
			for _, baseAlg := range baseLog.specIDEvent.algs {
				if baseAlg == alg {
					algOk = true
					break
				}
			}
			if !algOk {
				return nil, fmt.Errorf("log %d: cannot use digest (%+v) not present in base log", i, alg)
			}
		}
```